Sources/OktaIdx/IDXAuthenticator.swift

```diff
@@ -0,0 +1,382 @@
+//
+// Copyright (c) 2021, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+
+/// Protocol authenticators may conform to if they are capable of the "send" action.
+///
+/// This is often used in Phone authenticators.
+@objc(IDXAuthenticatorIsSendable)
+public protocol Sendable {
+    /// Determines if this action can perform the send action.
+    @objc var canSend: Bool { get }
+    
+    /// Sends the authentication code.
+    /// - Parameter completion: Completion handler when the response is returned, or `nil` if the developer does not need to handle the response.
+    @objc func send(completion: IDXClient.ResponseResult?)
```

Comment:
If completion is nullable how the developer will get know the result? Is there a delegate alternative?

Reply:
There are some circumstances where a developer doesn't care about the result, particularly with the "send" and "resend" scenarios. For example, if a user is trying to validate a Phone/SMS authenticator, and they can't get the code for some reason, they can ask for the code to be resent. From a UI perspective, they may not care about the response status of the request, since the code is going to be received out-of-band.  All they care is that clicking "send" will (hopefully) send a new code.

Alternatively, they can implement the delegate and be notified of the response that way.